lib/oban/registry.ex

```diff
@@ -0,0 +1,31 @@
+defmodule Oban.Registry do
+  @moduledoc false
+
+  @type role :: term
+  @type key :: Oban.name() | {Oban.name(), role}
+  @type value :: term
+
+  def child_spec(_arg) do
+    Supervisor.child_spec(
+      Registry.child_spec(keys: :unique, name: __MODULE__),
+      id: __MODULE__
+    )
+  end
+
+  @spec config(Oban.name()) :: Oban.Config.t()
+  def config(oban_name) do
+    [{_pid, config}] = Registry.lookup(__MODULE__, oban_name)
+    config
+  end
+
+  @spec whereis(Oban.name(), role) :: pid | nil
```

Comment:
`nil` is definitely not implied as it is a concrete type (atom of value `:nil`). Omitting it would mean that `whereis` will always return `pid`, so it is needed here. See also [GenServer.whereis spec](https://hexdocs.pm/elixir/GenServer.html#whereis/1).